src/pages/Account/Tabs/ModulesTab/Contract.tsx

```diff
@@ -450,11 +456,70 @@ function ReadContractForm({
 
   const onSubmit: SubmitHandler<ContractFormType> = async (data) => {
     logEvent("read_button_clicked", fn.name);
-    const viewRequest: Types.ViewRequest = {
-      function: `${module.address}::${module.name}::${fn.name}`,
-      type_arguments: data.typeArgs,
-      arguments: data.args,
-    };
+    let viewRequest: Types.ViewRequest;
+    try {
+      viewRequest = {
+        function: `${module.address}::${module.name}::${fn.name}`,
+        type_arguments: data.typeArgs,
+        arguments: data.args.map((arg, i) => {
+          if (fn.params[i].includes("vector")) {
+            // encode the vector
+            const rawVector = arg.trim().slice(1, -1).split(",");
+            const regex = /vector<([^>]+)>/;
+            const match = fn.params[i].match(regex);
+            if (match) {
+              let bcsVector: any[] = [];
+              // encode the element in vector
+              bcsVector = rawVector.map((v) => {
```

Comment:
can we create a util function and separate this code snippet out? this file size's already a bit overwhelming.

Reply:
done.